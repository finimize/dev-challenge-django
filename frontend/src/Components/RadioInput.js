import React, { Component } from 'react'
import PropTypes from 'prop-types'

export default class RadioInput extends Component {
  constructor(props) {
    super(props)

    this.state = {
      selectedOption: props.defaultValue
    }
  }

  handleChange({ target: { value } }) {
    this.setState({selectedOption: value})
    this.props.onChange(value)
  }

  render() {
    const {selectedOption} = this.state

    return (
      <form>
        <div>
          <label>
            <input
              type="radio"
              value="monthly"
              checked={selectedOption === "monthly"}
              onChange={this.handleChange.bind(this)}
            />
            monthly
          </label>
        </div>
        <div>
          <label>
            <input
              type="radio"
              value="quarterly"
              checked={selectedOption === "quarterly"}
              onChange={this.handleChange.bind(this)}
            />
             quarterly
          </label>
        </div>
        <div>
          <label>
            <input
              type="radio"
              value="yearly"
              checked={selectedOption === "yearly"}
              onChange={this.handleChange.bind(this)}
            />
            yearly
          </label>
        </div>
      </form>
    )
  }

}

RadioInput.propTypes = {
  defaultValue: PropTypes.string
}
