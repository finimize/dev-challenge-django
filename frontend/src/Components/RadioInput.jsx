import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class RadioInput extends Component {
  constructor(props) {
    super(props);

    this.state = {
      selectedOption: props.defaultValue,
    };
  }

  handleChange({ target: { value } }) {
    this.setState({ selectedOption: value });
    const { onChange } = this.props;
    onChange(value);
  }

  render() {
    const { selectedOption } = this.state;

    return (
      <form>
        <div>
          <label>
            <input
              type="radio"
              value="monthly"
              checked={selectedOption === 'monthly'}
              onChange={(e) => { this.handleChange(e); }}
            />
            monthly
          </label>
        </div>
        <div>
          <label>
            <input
              type="radio"
              value="quarterly"
              checked={selectedOption === 'quarterly'}
              onChange={(e) => { this.handleChange(e); }}
            />
             quarterly
          </label>
        </div>
        <div>
          <label>
            <input
              type="radio"
              value="yearly"
              checked={selectedOption === 'yearly'}
              onChange={(e) => { this.handleChange(e); }}
            />
            yearly
          </label>
        </div>
      </form>
    );
  }
}

RadioInput.propTypes = {
  defaultValue: PropTypes.string,
  onChange: PropTypes.func,
};

RadioInput.defaultProps = {
  defaultValue: 'yearly',
  onChange: () => {},
};
