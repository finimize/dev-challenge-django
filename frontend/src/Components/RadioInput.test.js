import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';

import RadioInput from './RadioInput';

describe('RadioInput', () => {
    it('renders correctly', () => {
        const tree = TestRenderer.create(
            <RadioInput value={5} onChange={() => {}}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
})
