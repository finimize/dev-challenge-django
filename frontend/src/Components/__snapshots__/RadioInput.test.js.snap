// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioInput renders correctly 1`] = `
<form>
  <div>
    <label>
      <input
        checked={false}
        onChange={[Function]}
        type="radio"
        value="monthly"
      />
      monthly
    </label>
  </div>
  <div>
    <label>
      <input
        checked={false}
        onChange={[Function]}
        type="radio"
        value="quarterly"
      />
      quarterly
    </label>
  </div>
  <div>
    <label>
      <input
        checked={false}
        onChange={[Function]}
        type="radio"
        value="yearly"
      />
      yearly
    </label>
  </div>
</form>
`;
